### PR TITLE
Importing additional Anime Types from Simkl

### PR DESCRIPTION
**Pull request by jtnguyen45**
#### Database Migration
NO

#### Description
Users were getting error messages when trying to import their anime watchlist if they had the any OVAs, ONAs, Special, or Music Video anime types included in their watchlist. Added enum entries in simkl API, as well as tweaked the import warning message to fix this issue.

#### Issues Fixed or Closed by this PR
Fixes https://github.com/Sonarr/Sonarr/issues/5839

### CODE REVIEW

File: src/NzbDrone.Core/ImportLists/Simkl/SimklAPI.cs [hunk 1]
```diff
@@ -75,6 +76,12 @@ public class SimklTvSyncActivityResource
     public enum SimklAnimeType
     {
         Tv,
-        Movie
```
Comment (markus101) [inline src/NzbDrone.Core/ImportLists/Simkl/SimklAPI.cs:78]: A value of `Unknown` above `Tv` would be best to capture all the types Sonarr doesn't know about, which will be used if it doesn't match a known type.

File: src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs [hunk 2]
```diff
@@ -43,7 +43,7 @@ public virtual IList<ImportListItemInfo> ParseResponse(ImportListResponse import
                 {
                     var tentativeTvdbId = int.TryParse(show.Show.Ids.Tvdb, out var tvdbId) ? tvdbId : 0;
 
-                    if (tentativeTvdbId > 0 && show.AnimeType == SimklAnimeType.Tv)
+                    if (tentativeTvdbId > 0 && show.AnimeType != SimklAnimeType.Movie && show.AnimeType != SimklAnimeType.Music)
```
Comment (markus101) [inline src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:46]: Leaving this to compare to Tv makes it easier to read and doesn't need to be updated every time a new value is added.

File: src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs [hunk 3]
```diff
@@ -55,7 +55,7 @@ public virtual IList<ImportListItemInfo> ParseResponse(ImportListResponse import
                     }
                     else
                     {
-                        Logger.Warn("Skipping info grabbing for '{0}' because it is a movie or it is not the first season of the show", show.Show.Title);
+                        Logger.Warn($"Skipping info grabbing for '{show.Show.Title}' because it is missing a TVDB ID, or it is movie or music video");
```
Comment (markus101) [inline src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:58]: Similar here, but instead of movie it should say something like "unsupported content type"

**Review by markus101 — CHANGES_REQUESTED** [review]

File: src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs (hunk 2 above)
Comment (jtnguyen45) [reply src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:46]: We were accounting for the different anime types ONAs, OVAs, and Specials if they had a TVDB id. Should we not include them? Cyberpunk: Edgerunners is listed as an ONA, for example.

Comment (markus101) [reply src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:46]: In that case opting into the ones we want is probably safer than excluding the ones we don't

Comment (jtnguyen45) [reply src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:46]: That makes sense. I will do that.

**Comment by jtnguyen45** [comment]
We added a [class](https://gist.github.com/gubenkoved/999eb73e227b7063a67a50401578c3a7) that helps handle AnimeTypes that are not defined in the SimklAnimeType enum. This should fix the import list bug.

File: src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs [hunk 4]
```diff
@@ -55,7 +55,7 @@ public virtual IList<ImportListItemInfo> ParseResponse(ImportListResponse import
                     }
                     else
                     {
-                        Logger.Warn("Skipping info grabbing for '{0}' because it is a movie or it is not the first season of the show", show.Show.Title);
+                        Logger.Warn($"Skipping info grabbing for '{0}' because it is an unsupported content type.", show.Show.Title);
```
Comment (markus101) [inline src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:58]: The `$` is unneeded.

File: src/NzbDrone.Core/ImportLists/Simkl/SimklAPI.cs (hunk 1 above)
Comment (markus101) [reply src/NzbDrone.Core/ImportLists/Simkl/SimklAPI.cs:78]: My mistake, I forgot that parsing that enum wouldn't work with the unknown values out of the box, but what you found will work.

File: src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs [hunk 5]
```diff
@@ -43,7 +43,7 @@ public virtual IList<ImportListItemInfo> ParseResponse(ImportListResponse import
                 {
                     var tentativeTvdbId = int.TryParse(show.Show.Ids.Tvdb, out var tvdbId) ? tvdbId : 0;
 
-                    if (tentativeTvdbId > 0 && show.AnimeType == SimklAnimeType.Tv)
+                    if (tentativeTvdbId > 0 && (show.AnimeType == SimklAnimeType.Tv || show.AnimeType == SimklAnimeType.Ona || show.AnimeType == SimklAnimeType.Ova || show.AnimeType == SimklAnimeType.Special))
```
Comment (markus101) [inline src/NzbDrone.Core/ImportLists/Simkl/SimklParser.cs:46]: With C# 9 you can now do the following which makes it a little easier to read and add more options if the need arises.

```suggestion
                    if (tentativeTvdbId > 0 && (show.AnimeType is SimklAnimeType.Tv or SimklAnimeType.Ona or SimklAnimeType.Ova or SimklAnimeType.Special))
```

**Review by markus101 — CHANGES_REQUESTED** [review]
Nice solution for the enum problem, just a couple small suggestions.

**Review by markus101 — APPROVED** [review]

**Pull request merged**